pandas/core/internals/managers.py

```diff
@@ -1027,6 +1027,8 @@ def iset(self, loc: Union[int, slice, np.ndarray], value):
         Set new item in-place. Does not consolidate. Adds new Block if not
         contained in the current set of items
         """
+        if isinstance(value, ABCSeries):
+            value = value._values
```

Comment:
@jbrockmendel I suppose ideally we ensure that `iset` always already gets the correct values, so never a Series. And so I could also unpack (or maybe use `extract_array`?) where `iset` is called:

https://github.com/pandas-dev/pandas/blob/a1f605697e44e7668b64841baf4ab2988470d08c/pandas/core/generic.py#L3273-L3278

But: 1) I am not fully sure in which case this `_maybe_cache_changed` is used, and so `value` can also be something else as a Series (so DataFrame)? and 2) `iset` is being used in other places as well, where the same issue might occur

Reply:
re 2: I think iset is only called from 2 places in NDFrame.  shouldnt be that hard to check unwrapping there, annotate iset

Reply:
Yeah, `iset` itself is indeed only additionally used in `_iset_item`, but that is then used in `_set_item` and in indexing code, ... (and checking in the indexing code what values can be passed is already a bit less trivial ;-))

Reply:
OK.  I might do a follow-up to try to push the check earlier